tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -1664,6 +1715,10 @@ int tracepoint__raw_syscalls__sys_enter(struct bpf_raw_tracepoint_args *ctx)
             bpf_map_update_elem(&new_pids_map, &pid, &pid, BPF_ANY);
         }
     }
+    else if (id == SYSCALL_CONNECT || id == SYSCALL_ACCEPT || id == SYSCALL_ACCEPT4 || id == SYSCALL_BIND || id == SYSCALL_LISTEN) {
+        u32 sockfd = args_tmp.args[0];
+        save_sockfd(sockfd);
+    }
```

Comment:
Why not putting this after `should_trace()`?

Reply:
because if someone wants to only trace for example 'security_socket_connect' then the 'connect' event wouldn't be traced. in this scenario, if i'll put this if statement after the sould_trace(), the 'sockfd' argument of 'security_socket_connect' will be missing.

Reply:
When two events depend on each other, we can set an "essential event" which will be traced, but not printed as part of the output. This happen, for example, when you choose magic_write event but not vfs_write, which is needed for magic_write to work. In this case, you can set the connect event if security_socket_connect is chosen. To do that, have a look at how we do this for magic_write in tracee.go, in the function New()